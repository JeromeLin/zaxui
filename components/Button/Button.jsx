import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Spinner from '../Spinner';

class Button extends PureComponent {
  render() {
    const { prefixCls, className, theme, size, shape, icon, block, active, focus, bordered, disabled, loading, tabIndex, onClick, children, ...others } = this.props;

    const classes = classnames({
      [`${prefixCls}`]: true,
      [className]: !!className,
      [`theme-${theme}`]: !!theme,
      [`size-${size}`]: !!size,
      [`shape-${shape}`]: !!shape,
      block,
      bordered,
      active,
      focus,
      disabled,
    });

    const iconRender = loading
      ? <Spinner className="rotate360" />
      : icon;

    const contentRender = (!!icon || loading)
      ? <div className={`${prefixCls}-content`}>{iconRender}<span>{children}</span></div>
      : children;

    return (
      <a
        {...others}
        role="button"
        tabIndex={tabIndex}
        aria-disabled={disabled}
        className={classes}
        onTouchStart={() => {}}
        onClick={e => !disabled && onClick(e)}>
        {contentRender}
      </a>
    );
  }
}

Button.propTypes = {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  theme: PropTypes.oneOf(['default', 'primary', 'info', 'success', 'warning', 'error']),
  size: PropTypes.oneOf(['xl', 'lg', 'sm', 'xs']),
  shape: PropTypes.oneOf(['radius', 'round', 'circle']),
  block: PropTypes.bool,
  bordered: PropTypes.bool,
  active: PropTypes.bool,
  focus: PropTypes.bool,
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
  tabIndex: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  onClick: PropTypes.func,
};

Button.defaultProps = {
  prefixCls: 'za-button',
  className: null,
  theme: 'default',
  size: null,
  shape: null,
  block: false,
  bordered: false,
  active: false,
  focus: false,
  disabled: false,
  loading: false,
  tabIndex: null,
  onClick() {},
};

export default Button;
