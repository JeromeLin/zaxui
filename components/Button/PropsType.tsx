export default interface PropsType {
  theme?: 'default' | 'primary' | 'info' | 'success' | 'warning' | 'error';
  size?: 'xl' | 'lg' | 'sm' | 'xs';
  shape?: 'radius' | 'round' | 'circle';
  block?: boolean;
  bordered?: boolean;
  active?: boolean;
  focus?: boolean;
  disabled?: boolean;
  loading?: boolean;
  icon?: any;
  onClick?: (x?: any) => void;
}

