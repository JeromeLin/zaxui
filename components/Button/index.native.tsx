import React, { PureComponent } from 'react';
import { StyleSheet, Text, View } from 'react-native';

const buttonStyles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
});

export default class Button extends PureComponent<{}> {
  static defaultProps = {
    prefixCls: 'za-button',
    theme: 'default',
    block: false,
    bordered: false,
    active: false,
    focus: false,
    disabled: false,
    loading: false,
    styles: buttonStyles,
    onClick() {},
  };

  render() {
    const {
      theme,
      size,
      shape,
      icon,
      block,
      active,
      focus,
      bordered,
      disabled,
      loading,
      onClick,
      styles,
      children,
    } = this.props;

    return (
      <View style={styles.container}>
        <Text>{children}</Text>
      </View>
    );
  }
}
