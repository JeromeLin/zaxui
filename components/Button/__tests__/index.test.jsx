import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../index';

describe('Button', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Button>foo</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
