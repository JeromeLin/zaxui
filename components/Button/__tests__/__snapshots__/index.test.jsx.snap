// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<a
  aria-disabled={false}
  className="za-button theme-default"
  onClick={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={null}
>
  <span>
    foo
  </span>
</a>
`;
